app/src/main/java/org/fossasia/openevent/app/data/sponsor/SponsorDelegateImpl.java

```diff
@@ -0,0 +1,71 @@
+package org.fossasia.openevent.app.data.sponsor;
+
+import android.databinding.DataBindingUtil;
+import android.support.annotation.NonNull;
+import android.view.View;
+
+import com.mikepenz.fastadapter.items.AbstractItem;
+
+import org.fossasia.openevent.app.R;
+import org.fossasia.openevent.app.common.model.HeaderProvider;
+import org.fossasia.openevent.app.core.sponsor.list.viewholder.SponsorsViewHolder;
+import org.fossasia.openevent.app.utils.CompareUtils;
+
+import java.util.List;
+
+public class SponsorDelegateImpl extends AbstractItem<Sponsor, SponsorsViewHolder> implements Comparable<Sponsor>, HeaderProvider {
+
+    private final Sponsor sponsor;
+
+    public SponsorDelegateImpl(Sponsor sponsor) {
+        this.sponsor = sponsor;
+    }
+
+    @Override
+    public int compareTo(@NonNull Sponsor other) {
+        return CompareUtils.compareCascading(sponsor, other, Sponsor::getName);
+    }
+
+    @Override
+    public long getIdentifier() {
+        return sponsor.getId();
+    }
+
+    @Override
+    public int getType() {
+        return 0;
+    }
+
+    @Override
+    public int getLayoutRes() {
+        return R.layout.sponsor_item;
+    }
+
+    @Override
+    public SponsorsViewHolder getViewHolder(@NonNull View view) {
+        return new SponsorsViewHolder(DataBindingUtil.bind(view));
+    }
+
+    @Override
+    public void bindView(@NonNull SponsorsViewHolder holder, @NonNull List<Object> list) {
+        super.bindView(holder, list);
+        holder.bindSponsor(sponsor);
+    }
+
+    @Override
+    public void unbindView(@NonNull SponsorsViewHolder holder) {
+        super.unbindView(holder);
+        holder.unbindSponsor();
+    }
+
+    @Override
+    public String getHeader() {
+        return sponsor.getName().substring(0, 1);
```

Comment:
Should be simply getName() in case you want to add name as header, if you want to add first character of name then it is fine. In my opinion we can give sponsorship tier as a header, do you agree @iamareebjamal

Reply:
Should be level